File: Ejercicio 5/empleados.java
class Empleado {
    String nombre;
    String documento;
    double salarioBase;
    String fechaIngreso;

    public Empleado(String nombre, String documento, double salarioBase, String fechaIngreso) {
        this.nombre = nombre;
        this.documento = documento;
        this.salarioBase = salarioBase;
        this.fechaIngreso = fechaIngreso;
    }

    public void mostrarInfo() {
        System.out.println("Nombre: " + nombre);
        System.out.println("Documento: " + documento);
        System.out.println("Salario Base: " + salarioBase);
        System.out.println("Fecha de ingreso: " + fechaIngreso);
    }

    public double calcularSalario() {
        return salarioBase;
    }
}


class Gerente extends Empleado {
    double bono;

    public Gerente(String nombre, String documento, double salarioBase, String fechaIngreso, double bono) {
        super(nombre, documento, salarioBase, fechaIngreso);
        this.bono = bono;
    }

    public double calcularSalario() {
        return salarioBase + bono;
    }
}


class Operario extends Empleado {
    int horas;
    double pagoHora;

    public Operario(String nombre, String documento, double salarioBase, String fechaIngreso,
                    int horas, double pagoHora) {
        super(nombre, documento, salarioBase, fechaIngreso);
        this.horas = horas;
        this.pagoHora = pagoHora;
    }

    public double calcularSalario() {
        return horas * pagoHora;
    }
}


class Desarrollador extends Empleado {
    String lenguaje;

    public Desarrollador(String nombre, String documento, double salarioBase, String fechaIngreso,
                         String lenguaje) {
        super(nombre, documento, salarioBase, fechaIngreso);
        this.lenguaje = lenguaje;
    }

    public double calcularSalario() {
        return salarioBase;
    }
}


public class Main {
    public static void main(String[] args) {

        Empleado gerente = new Gerente("Laura", "1001", 3000000, "2020", 1000000);
        Empleado operario = new Operario("Pedro", "2002", 0, "2022", 160, 10000);
        Empleado dev = new Desarrollador("Andres", "3003", 4000000, "2021", "Java");

        Empleado[] empleados = {gerente, operario, dev};

        for (Empleado e : empleados) {
            System.out.println("Tipo: " + e.getClass().getSimpleName());
            e.mostrarInfo();
            System.out.println("Salario total: " + e.calcularSalario());
            System.out.println("-------------------");
        }
    }
}

File: Ejercicio 5/empleados.py
class Empleado:
    def __init__(self, nombre, documento, salario_base, fecha_ingreso):
        self.nombre = nombre
        self.documento = documento
        self.salario_base = salario_base
        self.fecha_ingreso = fecha_ingreso

    def mostrar_info(self):
        print("Nombre:", self.nombre)
        print("Documento:", self.documento)
        print("Salario Base:", self.salario_base)
        print("Fecha ingreso:", self.fecha_ingreso)

    def calcular_salario(self):
        return self.salario_base


class Gerente(Empleado):
    def __init__(self, nombre, documento, salario_base, fecha_ingreso, bono):
        super().__init__(nombre, documento, salario_base, fecha_ingreso)
        self.bono = bono

    def calcular_salario(self):
        return self.salario_base + self.bono


class Operario(Empleado):
    def __init__(self, nombre, documento, salario_base, fecha_ingreso, horas, pago_hora):
        super().__init__(nombre, documento, salario_base, fecha_ingreso)
        self.horas = horas
        self.pago_hora = pago_hora

    def calcular_salario(self):
        return self.horas * self.pago_hora


class Desarrollador(Empleado):
    def __init__(self, nombre, documento, salario_base, fecha_ingreso, lenguaje):
        super().__init__(nombre, documento, salario_base, fecha_ingreso)
        self.lenguaje = lenguaje

    def calcular_salario(self):
        return self.salario_base



gerente = Gerente("Laura", "1001", 3000000, "2020", 1000000)
operario = Operario("Pedro", "2002", 0, "2022", 160, 10000)
dev = Desarrollador("Andres", "3003", 4000000, "2021", "Java")

empleados = [gerente, operario, dev]

for e in empleados:
    print("Tipo:", type(e).__name__)
    e.mostrar_info()
    print("Salario total:", e.calcular_salario())
    print("-------------------")
